src/lib.rs

```diff
@@ -295,7 +294,9 @@ impl ErasureCoder {
 }
 impl Drop for ErasureCoder {
     fn drop(&mut self) {
-        let _ = c_api::instance_destroy(self.desc);
+        with_global_lock(|| {
```

Comment:
Could you please add a short description which describes why we must use a global lock here?